changelog.d/20221104_145601_sirosen_add_scope_parsing.rst

```diff
@@ -15,3 +15,9 @@
     ``Scope(...).add_dependency("*foo")`` can be used to add an optional
     dependency as a string, or equivalently
     ``Scope(...).add_dependency(Scope("foo", optional=True))``
+
+  * ``Scope``\s support containment checking (``Scope("foo") in Scope("bar")``),
```

Comment:
I'm a bit torn. I'm actually not even sure I want us to implement this idea with `__contains__`.

Maybe this should be a set of comparators so that it's `scope1 < scope2`? That's more laborious to define, but maybe easier to explain.

Or we could stay away from the dunder methods and call this something like "contains()"?

Reply:
I think the only difference between `.contains()` vs `.__contains__()` is that we get to write `smaller_thing in maybe_bigger_thing`. If "smaller" and "bigger" aren't meaningful ways to think about this, it might not be intuitive to write `scope1 < scope2`.

Spit-balling just to get the gears turning:

* `.is_superset()`
* `.will_get()`
* `.overlays()`

Reply:
Per our discussion, we agreed on "permission coverage checking" as the textual name for this, but `__contains__` as the method to use such that `x in y` is described as "coverage checking".

I'll make sure that we get a docstring on `__contains__` which uses the same terminology, and that we include it in docs via the sphinx directive (I think `:special-members:`? but I'll double-check that).